src/lib/y2security/selinux_config.rb

```diff
@@ -0,0 +1,290 @@
+
+    Yast.import "Bootloader"
+    Yast.import "ProductFeatures"
+
+    # @return [SelinuxConfig::Mode] the last set mode, which can be differrent to the
+    #   {#running_mode} and {#configured_mode}. A call to {#save} is needed to make it the
+    #   {#configured_mode} for the next boot.
+    attr_reader :mode
+
+    # Constructor
+    def initialize
+      @mode = initial_mode
+    end
+
+    # Returns the initial SELinux mode
+    #
+    # It can be the proposed one if running in an installation or the configured mode for a
+    # running system
+    #
+    # @return [SelinuxConfig::Mode]
+    def initial_mode
+      @initial_mode ||= Yast::Mode.installation ? propose_mode : configured_mode
+    end
+
+    # Returns the mode applied in the running system
+    #
+    # @note the system can be booted with a different SELinux mode that the configured one. To
+    #  know the running mode getenforce tool is used.
+    #
+    # @return [Mode] running SELinux mode if command executed successfully; :disabled otherwise
+    def running_mode
+      id = Yast::Execute.locally!(GETENFORCE_PATH, stdout: :capture).chomp.downcase.to_sym
+      Mode.find(id)
+    rescue Cheetah::ExecutionFailed => e
+      log.info(e.message)
+
+      Mode.find(:disabled)
+    end
+
+    # Returns a collection holding all known SELinux modes
+    #
+    # @return [Array<SelinuxConfig::Mode>] a collection of known SELinux modes
+    def modes
+      Mode.all
+    end
+
+    # Set the mode to given value
+    #
+    # @note using nil means to set SELinux mode as disabled.
+    #
+    # @param id [SelinuxConfig::Mode, String, Symbol, nil] a SELinux mode or its identifier
+    # @return [Mode] the SelinuxConfig::Mode by given id or disabled is none found or nil was given
+    def mode=(id)
+      found_mode = Mode.find(id)
+
+      if found_mode.nil?
+        log.info("Requested SELinux mode `#{id}` not found. Falling back to :disabled.")
+        found_mode = Mode.find(:disabled)
+      end
+
+      @mode = found_mode
+      @mode
+    end
+
+    # Set current mode options as kernel parameters for the next boot
+    #
+    # @note it does not write the changes when running in installation mode, where only sets the
+    #   kernel params in memory since Yast::Bootloader.Write will be performed at the end of
+    #   installation.
+    #
+    # @see Yast::Bootloader#modify_kernel_params
+    #
+    # @return [Boolean] false if mode is not set or nothing changed; true otherwise
+    def save
+      Yast::Bootloader.modify_kernel_params(mode.options)
+
+      return true if Yast::Mode.installation
+
+      Yast::Bootloader.Write
+    end
+
+    private
+
+    # Path to the SELinux getenforce command
+    GETENFORCE_PATH = "/usr/sbin/getenforce".freeze
+    private_constant :GETENFORCE_PATH
+
+    # Proposes a mode based on `selinux_mode` value set in the control file
+    #
+    # @see Mode.find
+    #
+    # @return [Mode] disabled or found SELinux mode
+    def propose_mode
```

Comment:
I would call it `proposed_mode` to be consistent with configured mode and also makes more sense.